specifyweb/frontend/js_src/lib/components/DataModel/businessRules.ts

```diff
@@ -0,0 +1,375 @@
+import { overwriteReadOnly, RA } from '../../utils/types';
+import { AnySchema, AnyTree, TableFields } from './helperTypes';
+import { SpecifyResource } from './legacyTypes';
+import { BusinessRuleDefs, businessRuleDefs } from './businessRuleDefs';
+import { flippedPromise, ResolvablePromise } from '../../utils/promise';
+import { isTreeResource } from '../InitialContext/treeRanks';
+import { initializeTreeRecord, treeBusinessRules } from './treeBusinessRules';
+import { Collection } from './specifyModel';
+import { SaveBlockers } from './saveBlockers';
+import { formatConjunction } from '../Atoms/Internationalization';
+import { formsText } from '../../localization/forms';
+import { LiteralField, Relationship } from './specifyField';
+import { idFromUrl } from './resource';
+import { globalEvents } from '../../utils/ajax/specifyApi';
+
+var enabled: boolean = true;
+
+globalEvents.on('initResource', (resource) => {
+  enabled && !resource.noBusinessRules
+    ? attachBusinessRules(resource)
+    : undefined;
+});
+
+export function enableBusinessRules(e: boolean) {
+  return (enabled = e);
+}
+
+export class BusinessRuleMgr<SCHEMA extends AnySchema> {
```

Comment:
Rename `BusinessRuleMgr` to `BusinessRuleManager`

Reply:
Fixed in 7c4bb25